### DISCUSSION TITLE
How to deal with maps.

### DISCUSSION THREAD

**Discussion by tpoxa** [swaggest/jsonschema-go#115] [Q&A] [answered]
Hello. I've been using jsonschema's `CollectDefinitions` to ... collect definitions. 

After some processing, I want to add processed definitions to the resulting schema.

So I do it with:
```
WithExtraPropertiesItem("$defs", defs)
```

Where `devs` is a `map[string]jsonschema.Schema`

Well, all works fine except that each time definitions are sorted randomly. I guess because of the nature of the hash map.
Is there any other way to do that and keep the content of the schema consistent?

Thanks!

**Comment by vearutop** [comment, marked as answer]
you can use some of the ordered maps implementations, like https://github.com/wk8/go-ordered-map for example

but I would not recommend depending on maps ordering, since it does not have any semantics in JSON and at any point of further processing original order can be lost
